resolwe/flow/executors/__init__.py

```diff
@@ -122,6 +123,14 @@ def run(self, data_id, script, verbosity=1):
 
         self.data_id = data_id
 
+        # Fetch data instance to get any executor requirements.
+        if data_id != 'no_data_id':
```

Comment:
I also dislike this. We should not complicate production code, to support one weird test case. How about you fix [this test](https://github.com/genialis/resolwe/blob/master/resolwe/flow/tests/test_executors.py#L81) to refer to an actual data object?